tests/test_operation.py

```diff
@@ -1186,14 +1199,37 @@ def test_basis_state_decomposition(self, monkeypatch):
     def test_qubit_state_vector_decomposition(self, monkeypatch):
         """Test the decomposition of QubitStateVector calls the
         MottonenStatePreparation template"""
-        state = np.array([1/2, 1j/np.sqrt(2), 0, -1/2])
+        state = np.array([1 / 2, 1j / np.sqrt(2), 0, -1 / 2])
         wires = [0, 1]
         call_args = []
 
         # We have to patch MottonenStatePreparation where it is loaded
-        monkeypatch.setattr(qml.ops.qubit, "MottonenStatePreparation", lambda *args: call_args.append(args))
+        monkeypatch.setattr(
+            qml.ops.qubit, "MottonenStatePreparation", lambda *args: call_args.append(args)
+        )
         qml.QubitStateVector.decomposition(state, wires=wires)
 
         assert len(call_args) == 1
         assert np.array_equal(call_args[0][0], state)
         assert np.array_equal(call_args[0][1], wires)
+
+
+class TestChannel:
+    """Unit tests for the Channel class"""
+
+    def test_grad_method(self):
+        """Test that an exception is raised if a gradient method is set to analytic
+        as only finite difference or ``None`` is allowed at the moment. This can be updated
+        once we add gradient recipes for channels. """
+
+        class DummyOp(qml.operation.Channel):
+            r"""Dummy custom channel"""
+            num_wires = 1
+            num_params = 1
+            par_domain = "R"
+            grad_method = "A"
+
+        with pytest.raises(
+            AssertionError, match="Analytic gradients can not be used for quantum channels"
```

Comment:
(hopefully, this will help with passing the checks :) )
```suggestion
            ValueError, match="Analytic gradients can not be used for quantum channels"
```